### PR TITLE
Docs landing page redesign

### PR DESCRIPTION
**Pull request by scottybollinger**
This PR launches a redesigned [docs landing page](https://www.elastic.co/guide/index.html). It builds on the efforts of #2805 and adds additional styling and a bit of JavaScript to change the layout for the links at the bottom of the page to be in 2 columns and also to have a collapsable set of legacy links. 

Note that the PR preview does not have the heading fonts, as they are only accessible on pages hosted on the elastic.co domain, but they will be on the live site. Here is a screenhot the live site rendered with the fonts in place:

mockup-2023-12-08

### CODE REVIEW

**Review by kilfoyle — APPROVED** [review]
Wow. @scottybollinger this new landing page looks gorgeous!

File: extra/docs_landing.html [hunk 1]
```diff
@@ -39,347 +39,607 @@
...
+            </h3>
+            <p>Deploy instances of the Elastic Stack in the cloud, with the provider of your choice.</p>
+          </div>
+        </a>
+      </div>
+      <div class="col-md-4 col-12 mb-2">
+        <a class="no-text-decoration" href="en/elastic-stack/current/installing-elastic-stack.html">
+          <div class="card h-100">
+            <div class="card-border card-border-top"></div>
+            <span class="inline-block float-left icon mr-2" style="background-image: url('https://images.contentstack.io/v3/assets/bltefdd0b53724fa2ce/bltcb0e79820c355263/636c1905cbd6b84ecb4851b2/stack-logo-color-32px.png');">
```
Comment (leemthompo) [inline extra/docs_landing.html:576]:
```suggestion
            <span class="inline-block float-left icon mr-2" style="background-image: url('https://images.contentstack.io/v3/assets/bltefdd0b53724fa2ce/blt0090c6239e64faf8/logo-stack-32-color.svg');">
```

Logo is a little hazy, this SVG might be be a good substitute.

File: extra/docs_landing.html [hunk 2]
```diff
@@ -39,347 +39,607 @@
...
+            </h3>
+            <p>Download, install, and run Elastic products.</p>
+          </div>
+        </a>
+      </div>
+      <div class="col-md-4 col-12 mb-2">
+        <a class="no-text-decoration" href="https://docs.elastic.co/serverless">
+          <div class="card h-100">
+            <div class="card-border card-border-top"></div>
+            <span class="inline-block float-left icon mr-2" style="background-image: url('https://images.contentstack.io/v3/assets/bltefdd0b53724fa2ce/bltcb0e79820c355263/636c1905cbd6b84ecb4851b2/stack-logo-color-32px.png');">
```
Comment (leemthompo) [inline extra/docs_landing.html:589]:
```suggestion
            <span class="inline-block float-left icon mr-2" style="background-image: url('https://images.contentstack.io/v3/assets/bltefdd0b53724fa2ce/blt0090c6239e64faf8/logo-stack-32-color.svg');">
```

Same suggestion :)

**Review by KOTungseth — DISMISSED** [review]

File: extra/docs_landing.html (hunk 2 above)
Comment (scottybollinger) [reply extra/docs_landing.html]: Nice catch! Fixed here https://github.com/elastic/docs/pull/2847/commits/716d899ee8c266abf80016d2d67650b7e57f6816

File: extra/docs_landing.html (hunk 1 above)
Comment (scottybollinger) [reply extra/docs_landing.html]: Nice catch! Fixed here https://github.com/elastic/docs/pull/2847/commits/716d899ee8c266abf80016d2d67650b7e57f6816

**Review by glitteringkatie — DISMISSED** [review]
Scotty, this looks stellar!

**Review by dedemorton — DISMISSED** [review]
LGTM. Feedback from Obs PM (Evin) was positive: "...the design and layout are far cleaner and more understandable, at least from my perspective."

**Pull request merged**